ui/components/app/modals/export-private-key-modal/export-private-key-modal.component.js

```diff
@@ -13,214 +19,309 @@ import {
           >
-            {this.context.t('done')}
+            {t('done')}
           </Button>
         ) : (
           <Button
+            type={BUTTON_TYPES.PRIMARY}
+            size={BUTTON_SIZES.LG}
+            width={BLOCK_SIZES.HALF}
             onClick={() => {
-              this.context.trackEvent({
-                category: EVENT.CATEGORIES.KEYS,
-                event: EVENT_NAMES.KEY_EXPORT_REQUESTED,
-                properties: {
-                  key_type: EVENT.KEY_TYPES.PKEY,
+              trackEvent(
+                {
+                  category: EVENT.CATEGORIES.KEYS,
+                  event: EVENT_NAMES.KEY_EXPORT_REQUESTED,
+                  properties: {
+                    key_type: EVENT.KEY_TYPES.PKEY,
+                  },
                 },
-              });
+                {},
+              );
 
-              this.exportAccountAndGetPrivateKey(this.state.password, address);
+              exportAccountAndGetPrivateKey(password, address);
             }}
-            type="primary"
-            large
-            className="export-private-key-modal__button"
-            disabled={!this.state.password}
+            disabled={!password}
           >
-            {this.context.t('confirm')}
+            {t('confirm')}
           </Button>
         )}
-      </div>
+      </Box>
     );
-  }
+  };
 
-  render() {
-    const {
-      selectedIdentity,
-      warning,
-      showAccountDetailModal,
-      hideModal,
-      previousModalState,
-    } = this.props;
-    const { name, address } = selectedIdentity;
+  const { name, address } = selectedIdentity;
 
-    const { privateKey, showWarning } = this.state;
+  return (
+    <AccountModalContainer
+      className="export-private-key-modal"
+      selectedIdentity={selectedIdentity}
+      showBackButton={previousModalState === 'ACCOUNT_DETAILS'}
+      backButtonAction={() => showAccountDetailModal()}
+    >
+      {showHoldToReveal ? (
+        <HoldToRevealModal
+          onLongPressed={() => setShowHoldToReveal(false)}
+          willHide={false}
+        />
+      ) : (
+        <>
+          <Text
+            as="span"
+            marginTop={2}
+            variant={TextVariant.bodyLgMedium}
+            fontWeight={FONT_WEIGHT.NORMAL}
+          >
+            {name}
+          </Text>
+          <Box
+            className="ellip-address-wrapper"
+            borderStyle={BorderStyle.solid}
+            borderColor={BorderColor.borderDefault}
+            borderWidth={1}
+            marginTop={2}
+            padding={[1, 2, 1, 2]}
+          >
+            {toChecksumHexAddress(address)}
+          </Box>
+          <Box
+            className="export-private-key-modal__divider"
+            width={BLOCK_SIZES.FULL}
+            margin={[5, 0, 3, 0]}
+          />
+          {/* children issue on types */}
+          {/* eslint-disable-next-line @typescript-eslint/ban-ts-comment */}
+          {/* @ts-ignore */}
```

Comment:
Can you explain a bit more why this is necessary?

Reply:
sorry, this was from migrating to tsx, but i've decided to split that into another pr. i've removed these